src/napari_topostats/_io.py

```diff
@@ -124,31 +125,41 @@ def build_dynamic_widget(
     return Container(widgets=widgets)
 
 
-@magicgui(
-    config_path={
-        "label": "Config file",
-        "mode": "r",
-        "filter": "*.yaml;*.json",
-    },  # Added .json filter
-    call_button="Load Config",
-    auto_call=True,
-)
-def load_config(viewer: Viewer, config_path: Path | None = None):
+def _load_config_impl(
+    viewer: Viewer, config_path: Path | None = None, use_default: bool = False
+):
     """
     Load a configuration file and build a dynamic widget to edit it.
     This is a magicgui function that can be called directly from the napari GUI and is an example of a hardcoded
     function being implemented using the dynamic function widget system.
     """
     global comment_descriptions, config_wrapper, full_config_container  # Updated global name
     if config_path is None:
-        args = Namespace()
-        args.config = None
-        args.filename = "_generated_config.yaml"
-        args.output_dir = None
-        args.simple = False  # This is to allow backwards compatibility with old version of topostatsdir
-        args.module = "topostats"
-        write_config_with_comments(args)
-        config_path = Path("_generated_config.yaml")
+        if use_default:
+            config_dir = Path(user_config_dir("TopoStats", "Napari"))
+            config_path = config_dir / "config.yaml"
+            if not config_path.exists():
+                args = Namespace()
+                args.config = None
+                args.filename = "_generated_config.yaml"
+                args.output_dir = config_dir
+                args.simple = False  # This is to allow backwards compatibility with old version of topostatsdir
+                args.module = "topostats"
+                write_config_with_comments(args)
+                config_path = Path(config_dir / args.filename)
+        else:
+            file_path, _ = QFileDialog.getOpenFileName(
+                parent=None,
+                caption="Select Config File",
+                filter="YAML Files (*.yaml *.yml);;JSON Files (*.json)",
```

Comment:
I'm pretty sure TopoStats doesn't support `.json` configuration files. Is there something in Napari that converts `.json` > `.yaml` that means they can be loaded?